GliaWidgets/SecureConversations/SecureConversations.Availability.swift

```diff
@@ -6,61 +6,118 @@ extension SecureConversations {
 
         var environment: Environment
 
-        func checkSecureConversationsAvailability(completion: @escaping CompletionResult) {
+        func checkSecureConversationsAvailability(
+            for queueIds: [String],
+            completion: @escaping CompletionResult
+        ) {
             environment.listQueues { queues, error in
                 if let error = error {
                     completion(.failure(error))
                     return
                 }
 
-                self.checkQueues(queues, completion: completion)
+                self.checkQueues(
+                    queueIds: queueIds,
+                    fetchedQueues: queues,
+                    completion: completion
+                )
             }
         }
 
         private func checkQueues(
-            _ queues: [CoreSdkClient.Queue]?,
+            queueIds: [String],
+            fetchedQueues: [CoreSdkClient.Queue]?,
             completion: (Result<Status, CoreSdkClient.SalemoveError>) -> Void
         ) {
             guard environment.isAuthenticated() else {
+                environment.log.warning("Message Center Unavailable because the visitor is unauthenticated.")
                 completion(.success(.unavailable(.unauthenticated)))
                 return
             }
 
-            guard let queues = queues, isSecureConversationsAvailable(in: queues) else {
+            guard let queues = fetchedQueues else {
+                // If no queue fetched from the server,
+                // return `.unavailable(.emptyQueue)`
                 completion(.success(.unavailable(.emptyQueue)))
                 return
             }
 
-            completion(.success(.available))
-        }
+            // if provided queueIds array contains invalid ids,
+            // then log a warning message.
+            let invalidIds = queueIds.filter { UUID(uuidString: $0) == nil }
+            if !invalidIds.isEmpty {
+                environment.log.warning("Queue ID array for Message Center contains invalid Queue IDs: \(invalidIds).")
+            }
 
-        private func isSecureConversationsAvailable(
-            in queues: [CoreSdkClient.Queue]
-        ) -> Bool {
-            let filteredQueues = queues
-                .filter {
-                    self.environment.queueIds.contains($0.id) &&
-                    $0.state.status != .closed &&
-                    $0.state.media.contains(CoreSdkClient.MediaType.messaging)
+            guard !queueIds.isEmpty else {
+                // If no queueId is passed, then check default queues instead
+                let defaultQueues = queues.filter(\.isDefault)
+                // Filter queues supporting `messaging` and are not `closed`
+                let filteredQueues = defaultQueues.filter(defaultPredicate)
+
+                if filteredQueues.isEmpty {
+                    environment.log.warning("No default queues supporting messaging and are not closed were found.")
```

Comment:
In that case `not closed` means queues that have status other than `closed`. Because it can have `open`, `closed`, `unstuffed`, `full` and `unknown`. So it should be `queue.status != .closed`

Would the next wording be better:
`No default queues that have status other than closed and support messaging were found.`

Reply:
Yeah, this sounds perfect! 👍

Reply:
Fixed 835c7460c4393daf1f1468f8af7d3753b15fed0f